relay-clients/client-asset-hub-rococo/src/lib.rs

```diff
@@ -0,0 +1,131 @@
+// Copyright 2022 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges Common.
+
+// Parity Bridges Common is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity Bridges Common is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges Common.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Types used to connect to the AssetHub-Rococo-Substrate parachain.
+
+pub mod codegen_runtime;
+
+use bp_asset_hub_rococo::AVERAGE_BLOCK_INTERVAL;
+use bp_polkadot_core::{SuffixedCommonTransactionExtension, SuffixedCommonTransactionExtensionExt};
+use codec::Encode;
+use relay_substrate_client::{
+	Chain, ChainWithBalances, ChainWithMessages, ChainWithRuntimeVersion, ChainWithTransactions,
+	Error as SubstrateError, SignParam, SimpleRuntimeVersion, UnderlyingChainProvider,
+	UnsignedTransaction,
+};
+use sp_core::{storage::StorageKey, Pair};
+use sp_runtime::{
+	generic::SignedPayload,
+	traits::{FakeDispatchable, IdentifyAccount},
+};
+use std::time::Duration;
+
+pub use codegen_runtime::api::runtime_types;
+use runtime_types::frame_metadata_hash_extension::Mode;
+
+use bp_runtime::extensions::{
+	BridgeRejectObsoleteHeadersAndMessages, GenericTransactionExtensionSchema,
+	RefundBridgedParachainMessagesSchema,
+};
+
+pub type CheckMetadataHash = GenericTransactionExtensionSchema<Mode, Option<[u8; 32]>>;
+
+pub type TransactionExtension = SuffixedCommonTransactionExtension<(
+	BridgeRejectObsoleteHeadersAndMessages,
+	RefundBridgedParachainMessagesSchema,
+	CheckMetadataHash,
```

Comment:
this needs to be aligned with asset-hub-rococo runtime: https://github.com/paritytech/polkadot-sdk/pull/6675/files#diff-a3b656e8095fe815cf8611c8319aff7b2f32482cdbe7c24ac37a2ce45235f32dR1173-R1174

```suggestion
pub type TransactionExtension = SuffixedCommonTransactionExtension<(
	CheckMetadataHash,
	BridgeRejectObsoleteHeadersAndMessages,
	RefundBridgedParachainMessagesSchema,
```